File: Algorithm/Leetcode/0012.py
# 1603. Design Parking System

class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.space = [0 , big, medium, small]        

    def addCar(self, carType: int) -> bool:
        self.space[carType] -= 1
        return 0 <= self.space[carType]

File: Algorithm/Leetcode/0000_0100/0011_0020/0012.cpp
// 1603. Design Parking System

class ParkingSystem {
private:
    array<int,4> spaces;
public:
    ParkingSystem(int big, int medium, int small)
        :spaces({0, big, medium, small})
    {}    
    bool addCar(int carType) 
    {
        return 0 < spaces[carType]--;
    }
};